File: withJava/src/practiceInterview/codility/L2_2.java
package practiceInterview.codility;

public class L2_2 {
  
  class Solution {
    public int[] solution(int[] A, int K) {
        int last = A.length - 1;
        int[] arr = new int[A.length];
        int i = 0;
        for(; i <= K; i++) {
            arr[K - i] = A[last - i];
        }
        
        if(K < last) {
            for(int j = i; j <= last; j++) {
                arr[j] = A[last - j];
            }
        }
        
        return arr;
    }
}

}

File: withJavascript/codility/L2_2.js



function solution(A, K) {
    var last = A.length - 1,
        arr = [],
        r = K > A.length ? K % A.length : K

    for(var i = last; r - (last - i) > 0; i--) {
        arr[r - (last - i) - 1] = A[i];
    }

    if(r < A.length) {
        for(var j = 0; j < A.length - r; j++) {
            arr[r + j] = A[j];
        }
    }

    return arr;
}
